artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/connect/mirror/AMQPMirrorControllerSource.java

```diff
@@ -357,6 +364,13 @@ private static void setProtocolData(MessageReference ref, String brokerID, long
             daMap.put(INTERNAL_DESTINATION, ref.getMessage().getAddress());
          }
       }
+
+      if (routingContext != null && routingContext.isMirrorIndividualRoute()) {
+         ArrayList<String> queues = new ArrayList<>();
+         routingContext.forEachDurable(q -> queues.add(String.valueOf(q.getName())));
+         daMap.put(TARGET_QUEUES, queues);
+      }
```

Comment:
non durable queues are not passed along on clustering. this is just for durable really.

The Mirror Individual Route is to avoid sending a message to an entire fan out when clustering redistribution happens.

Say you had a clustered message redistribution from one particular queue to another queue... which is a clustering operation. in that case we need to pass to the specific queues.

Reply:
So it should send the empty list if there are no durable queues? Or does that mean it shouldnt even send the message?

Reply:
sending it empty wouldn't be wrong.

The only scenario where this could happen is if the queue was removed in flight... 

Test this will be a bit complicated though.

Reply:
I would need to test on an impossible situation. That is an artificial operation routing to a local non durable queue. 

Clustered routings will always happen after durable queues.

Reply:
Ok, sounds like its fine then, just wanted to check, as its not obvious...very hard to tell stuff like that from the context if you dont already know it.